server/src/test/java/org/apache/cassandra/sidecar/handlers/GossipUpdateHandlerTest.java

```diff
@@ -0,0 +1,164 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
+import com.google.inject.util.Modules;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.apache.cassandra.sidecar.TestModule;
+import org.apache.cassandra.sidecar.cluster.CassandraAdapterDelegate;
+import org.apache.cassandra.sidecar.cluster.InstancesMetadata;
+import org.apache.cassandra.sidecar.cluster.instance.InstanceMetadata;
+import org.apache.cassandra.sidecar.common.server.StorageOperations;
+import org.apache.cassandra.sidecar.modules.SidecarModules;
+import org.apache.cassandra.sidecar.server.Server;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test for {@link GossipUpdateHandler}
+ */
+@ExtendWith(VertxExtension.class)
+public class GossipUpdateHandlerTest
+{
+    static final Logger LOGGER = LoggerFactory.getLogger(GossipUpdateHandlerTest.class);
+    Vertx vertx;
+    Server server;
+    StorageOperations mockStorageOperations = mock(StorageOperations.class);
+
+    @BeforeEach
+    void before() throws InterruptedException
+    {
+        Injector injector;
+        Module testOverride = Modules.override(new TestModule()).with(new GossipUpdateHandlerTest.GossipUpdateHandlerTestModule());
+        injector = Guice.createInjector(Modules.override(SidecarModules.all()).with(testOverride));
+        vertx = injector.getInstance(Vertx.class);
+        server = injector.getInstance(Server.class);
+        VertxTestContext context = new VertxTestContext();
+        server.start().onSuccess(s -> context.completeNow()).onFailure(context::failNow);
+        context.awaitCompletion(5, TimeUnit.SECONDS);
+    }
+
+    @AfterEach
+    void after() throws InterruptedException
+    {
+        CountDownLatch closeLatch = new CountDownLatch(1);
+        server.close().onSuccess(res -> closeLatch.countDown());
+        if (closeLatch.await(60, TimeUnit.SECONDS)) LOGGER.info("Close event received before timeout.");
+        else LOGGER.error("Close event timed out.");
+    }
+
+    @Test
+    void testStartGossip(VertxTestContext ctx)
+    {
+        WebClient client = WebClient.create(vertx);
+        String payload = "{\"state\":\"start\"}";
+        client.put(server.actualPort(), "127.0.0.1", "/api/v1/cassandra/gossip").expect(ResponsePredicate.SC_OK).sendBuffer(io.vertx.core.buffer.Buffer.buffer(payload), ctx.succeeding(resp -> {
```

Comment:
- This line is too long. If you run `./gradlew codeCheckTasks`, it would complain the line is too long. 
- `expect(ResponsePredicate.SC_OK)` is deprecated, can you update to its successor. 

The same applies to the other test assertions in this file and `NativeUpdateHandlerTest`.

Reply:
Agreed. I like the idea. We should spend less time on arguing about code style.